examples/batch/batch.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pgx "github.com/jackc/pgx/v5"
+	pgxpool "github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PgxIface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Close()
+}
+
+func databaseSetup(db PgxIface) (err error) {
+
+	// Initialize a database object
+	tx, err := db.Begin(context.Background())
+	if err != nil {
+		return fmt.Errorf("databaseSetup: %s", err)
+	}
+	// Finally, commit changes or rollback
+	defer func() {
+		switch err {
+		case nil:
+			err = tx.Commit(context.Background())
+		default:
+			_ = tx.Rollback(context.Background())
+		}
+	}()
+
+	// Create a new table 'ledger'
+	sql := `CREATE TABLE IF NOT EXISTS ledger (
+		id SERIAL PRIMARY KEY,
+		description VARCHAR(255) NOT NULL,
```

Comment:
it's better to use `VARCHAR` or `TEXT` without length specifier

Read [more](https://wiki.postgresql.org/wiki/Don't_Do_This#Don.27t_use_varchar.28n.29_by_default)